### PR TITLE
Fixing and enabling some Projects tests

### PR DESCRIPTION
**Pull request by Benjin**
STS update changes:
image

The code in `project.addScriptItem()` doesn't join mixed-slash paths properly, but since I'm intending to introduce `content` params to let DacFx.Projects handle new files and deprecate `addScriptItem()` in an upcoming PR (it's currently not used in the oracle migration extension), I figured I'd update the test and let the code removal take care of the bug.

### CODE REVIEW

File: extensions/sql-database-projects/src/test/projectController.test.ts [hunk 1]
```diff
@@ -597,7 +598,7 @@ describe('ProjectsController', function (): void {
 			should(opened).equal(true);
 		});
 
-		it.skip('Callbacks are hooked up and called from create project from database dialog', async function (): Promise<void> {
+		it('Callbacks are hooked up and called from create project from database dialog', async function (): Promise<void> {
```
Comment (kisantia) [inline extensions/sql-database-projects/src/test/projectController.test.ts:601]: this test was still failing in the PR validation

**Review by alanrenmsft — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

**Review by SakshiS-harma — APPROVED** [review]
What did the STS update changed, that the tests are passing now?

**Comment by Benjin** [comment]
@SakshiS-harma
> What did the STS update changed, that the tests are passing now?

There was yet another cross-plat path bug in the Projects library, fixed by https://msdata.visualstudio.com/SQLToolsAndLibraries/_git/DacFx/pullrequest/1073966.

File: extensions/sql-database-projects/src/test/projectController.test.ts (hunk 1 above)
Comment (Benjin) [reply extensions/sql-database-projects/src/test/projectController.test.ts]: I'm going to leave this one as skipped, and fix it separately.  I'm hitting something weird with the UI framework.

**Pull request merged**